### PR TITLE
Client implementation for 'transit secret' backend of Vault

### PR DESCRIPTION
**Pull request by innovationhub-asia**
- New client
- Bumped few dependencies
- Minor refactoring

### CODE REVIEW

**Comment by tlisonbee** [comment]
Hi @innovationhub-asia, thanks for the pull request!

There are some changes we'd need before we could consider merging but also you've reached us at an interesting point in time.  We're actually going to stop maintaining this client soon (within the next month or so).  

So rather than go through the review / merge process, I think what might make more sense is if you forked our repo.  The upside for you is you can make any changes you want.  There isn't really any downsides since this repo is going to be deprecated soon.  

We'd even be happy to link to your repo after we deprecate, if it was going to be maintained.

What do you think?

File: build.gradle [hunk 1]
```diff
@@ -20,23 +20,54 @@ buildscript {
...
+}
 
 task copyProjectVersion() {
     def releaseVersion = version
     doLast {
         ant.replace(file: "$buildDir/resources/main/java-vault-client.properties", token: "@@VAULT_CLIENT_RELEASE@@", value: releaseVersion)
     }
 }
+
+task resolveAllDependencies {
```
Comment (fieldju) [inline build.gradle:57]: What is the purpose of this block?

File: src/main/java/com/nike/vault/client/model/VaultCreateKeyRequest.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
+package com.nike.vault.client.model;
+
+import com.google.gson.annotations.SerializedName;
```
Comment (fieldju) [inline src/main/java/com/nike/vault/client/model/VaultCreateKeyRequest.java]: Remove this unused import.

**Comment by fieldju** [comment]
There are a couple unused imports floating around in various classes.

**Comment by fieldju** [comment]
Is there a way to not have circleci and travis config in the same project?

I think if you needed circle ci for your personal work flow, you could make a gradle multi-module project that used a git submodule to point to this repo

https://git-scm.com/book/en/v2/Git-Tools-Submodules

and then in your circle ci config you could do something like

`- run: ./gradlew sub-module:publishToGithub`

I haven't personally tried something like that before, but it sounds like it might work and be able to keep that config separate from the main project.

**Comment by fieldju** [comment]
@tlisonbee I am ok working through this PR and getting it merged, lets post on it after standup.

File: build.gradle [hunk 3]
```diff
@@ -20,23 +20,54 @@ buildscript {
...
 apply plugin: 'maven'
-apply plugin: 'com.jfrog.bintray'
 apply plugin: 'maven-publish'
+apply plugin: 'idea'
+apply plugin: 'git-repo'
 
-sourceCompatibility = 1.7
-targetCompatibility = 1.7
+sourceCompatibility = 1.8
+targetCompatibility = 1.8
```
Comment (fieldju) [inline build.gradle:28]: Why break compatibility with Java 7 users?

**Comment by fieldju** [comment]
We removed Vault from Cerberus, please closing this PR as I don't think it is relevant anymore. Please re-open if you disagree.

**Pull request closed without merging**